### PR TITLE
refactor query fetch and add position data

### PR DESCRIPTION
**Pull request by spclu6**
#### What problem is this solving?

This pr tries to solve the following:

- Uses the hook useDevice of the app 'vtex.device-detector' which returns more accurately what type of device the user is on than the one currently in use.

- It avoids the Graphql query to be performed every time the component is rendered thus avoiding to make calls before it loads the minimum information to obtain the personalized images.

- It refactors the way in which the latitude and longitude of the user can be obtained extracting that logic in a custom hook so that the Image component is cleaner.

#### How to test it?

[Workspace](https://imageprotocol--spclusellas.myvtex.com/?__bindingAddress=spclusellas.myvtex.com/)

#### Describe alternatives you've considered, if any.

An alternative that is being explored is to directly obtain the user's location through the ip in the backend and thus avoid waiting time on the client.

### CODE REVIEW

File: react/components/InfoCard/index.js [hunk 1]
```diff
@@ -141,32 +136,40 @@ const InfoCard = ({
...
-  if (!error && !loading && data && data.getImage && data.getImage.url !== null && data.getImage.urlMobile !== null && imageProtocolId !== '') {
-    console.log('getImage InfoCard: ',data.getImage)
-    imageHref = formatIOMessage({ id: data.getImage.hrefImg , intl })
-    actionHref = formatIOMessage({ id: data.getImage.hrefImg , intl })
-    if(mobile){
-      finalImageUrl = formatIOMessage({ id: data.getImage.urlMobile, intl })
+  const [getPersonalizedImages, { data: imageData }] = useLazyQuery(
+    GET_IMAGE_PROTOCOL_IMAGES,
+    {
+      ssr: false,
```
Comment (tomymehdi) [inline react/components/InfoCard/index.js:145]: why are we forcing the client side rendering?

File: react/hooks/usePosition.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react'
+
+interface Coordinate {
```
Comment (tomymehdi) [inline react/hooks/usePosition.ts:3]: let's replace `Coordinate` for `LatLong`. `Coordinate` is confusing

File: react/hooks/usePosition.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  latitude: number | undefined
+  longitude: number | undefined
+}
+
+interface Position {
```
Comment (tomymehdi) [inline react/hooks/usePosition.ts:8]: Maybe something like `LatLongRecord`, `LatLongHistory` or `LatLongTime` is a better fit here

**Review by tomymehdi — CHANGES_REQUESTED** [review]

**Review by LeslieJRP — APPROVED** [review]
changes reviewed

**Pull request merged**